File: accounting-system-web-project/nexa-erp-frontend/src/app/features/dashboard/services/dashboard.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { APP_CONFIG } from '../../../core/config/app.config';
import { ApiResponse } from '../../../core/models/api-response.model';
import { DashboardSummary, DashboardWorkflowSummary } from '../models/dashboard.model';

@Injectable({
  providedIn: 'root',
})
export class DashboardService {
  private readonly baseUrl = `${APP_CONFIG.apiUrl}/dashboard`;

  constructor(private http: HttpClient) {}

  getSummary(): Observable<ApiResponse<DashboardSummary>> {
    return this.http.get<ApiResponse<DashboardSummary>>(`${this.baseUrl}/summary`);
  }

  getWorkflowSummary(): Observable<ApiResponse<DashboardWorkflowSummary>> {
    return this.http.get<ApiResponse<DashboardWorkflowSummary>>(
      `${this.baseUrl}/workflow-summary`,
    );
  }
}

File: backend-for-all/backend/src/main/java/com/nexaerp/dashboard/DashboardService.java
package com.nexaerp.dashboard;

import com.nexaerp.dashboard.dto.DashboardSummaryDto;
import com.nexaerp.dashboard.dto.DashboardWorkflowSummaryDto;

public interface DashboardService {
    DashboardSummaryDto getSummary();
    DashboardWorkflowSummaryDto getWorkflowSummary();
}
